modules/tinymce/src/core/main/ts/delete/MergeBlocks.ts

```diff
@@ -55,10 +55,30 @@ const nestedBlockMerge = (
 
 const sidelongBlockMerge = (rootNode: SugarElement<Node>, fromBlock: SugarElement<Element>, toBlock: SugarElement<Element>): Optional<CaretPosition> => {
   if (Empty.isEmpty(toBlock)) {
-    Remove.remove(toBlock);
     if (Empty.isEmpty(fromBlock)) {
-      PaddingBr.fillWithPaddingBr(fromBlock);
+      const getInlineToBlockDescendants = (el: SugarElement<Element>) => {
+        const helper = (node: SugarElement<Element>, elements: SugarElement<Element>[]): SugarElement<Element>[] =>
+          Traverse.firstChild(node).fold(
+            () => elements,
+            (child) => ElementType.isInline(child) ? helper(child, elements.concat(Replication.shallow(child))) : elements
```

Comment:
I thought I followed up but clearly not (I've had a crazy day). Why did you re-add the `Replication.shallow`? I don't mind if we move the elements, or was there a bug this resolved?

Reply:
Since we've decided to filter all block type elements just moving the child throws `The new child is an ancestor of the parent`, which is understandable, since the element still contains its children. With `Replication.shallow` we map all the descendants, filter block type elements and making a shallow copy without children.